clone.go

```diff
@@ -0,0 +1,192 @@
+	var currentOverhang Overhang
+	var nextOverhang Overhang
+	if len(overhangs) == 1 { // Check the case of a single cut
+	}
+	for i := 0; i < len(overhangs)-1; i++ {
+		currentOverhang = overhangs[i]
+		nextOverhang = overhangs[i+1]
+		switch {
+		case (seq.Circular == true) && (enzyme.Directional == true):
+			if nextOverhang.Position > len(seq.Sequence) {
+				if (currentOverhang.Forward == true) && (nextOverhang.Forward == false) {
+					fragmentSeqs = append(fragmentSeqs, sequence[currentOverhang.Position:nextOverhang.Position])
+					break
+				}
+			}
+		case (seq.Circular == false) && (enzyme.Directional == true):
+			if (currentOverhang.Forward == true) && (nextOverhang.Forward == false) {
+				fragmentSeqs = append(fragmentSeqs, sequence[currentOverhang.Position:nextOverhang.Position])
+			}
+		case enzyme.Directional == false:
+			fragmentSeqs = append(fragmentSeqs, sequence[currentOverhang.Position:nextOverhang.Position])
+		}
+
+	}
+
+	// Convert fragment sequences into fragments
+	var fragments []Fragment
+	for _, fragment := range fragmentSeqs {
+		fragments = append(fragments, Fragment{Sequence: fragment[enzyme.EnzymeOverhangLen : len(fragment)-enzyme.EnzymeOverhangLen], ForwardOverhang: fragment[:enzyme.EnzymeOverhangLen], ReverseOverhang: fragment[len(fragment)-enzyme.EnzymeOverhangLen:]})
+	}
+	return fragments
+}
+
+func recurseLigate(wg *sync.WaitGroup, c chan string, seedFragment Fragment, fragmentList []Fragment) {
+	defer wg.Done()
+	if seedFragment.ForwardOverhang == seedFragment.ReverseOverhang {
+		c <- seedFragment.ForwardOverhang + seedFragment.Sequence
+	} else {
+		for _, newFragment := range fragmentList {
+			if seedFragment.ReverseOverhang == newFragment.ForwardOverhang {
+				newSeed := Fragment{seedFragment.Sequence + seedFragment.ReverseOverhang + newFragment.Sequence, seedFragment.ForwardOverhang, newFragment.ReverseOverhang}
+				wg.Add(1)
+				go recurseLigate(wg, c, newSeed, fragmentList)
+			}
+			if (seedFragment.ReverseOverhang == ReverseComplement(newFragment.ReverseOverhang)) && (seedFragment.ReverseOverhang != ReverseComplement(seedFragment.ReverseOverhang)) { // If the second statement isn't there, program will crash on palindromes
+				newSeed := Fragment{seedFragment.Sequence + seedFragment.ReverseOverhang + ReverseComplement(newFragment.Sequence), seedFragment.ForwardOverhang, ReverseComplement(newFragment.ForwardOverhang)}
+				wg.Add(1)
+				go recurseLigate(wg, c, newSeed, fragmentList)
+			}
+		}
+	}
+}
+
+func Ligate(fragments []Fragment, maxClones int) []CloneSequence {
+	var wg sync.WaitGroup
+	c := make(chan string, maxClones) // A buffered channel is needed to prevent blocking.
+	wg.Add(len(fragments))
+	for _, fragment := range fragments {
+		go recurseLigate(&wg, c, fragment, fragments)
+	}
+	wg.Wait()
+	close(c)
+
+	// For any given cloning reaction, there will be rotations that fulfill the requirement. This removes
+	// Rotations. Later, will be replaced with the seqhash algorithm
```

Comment:
Should this be replaced with seqhash now? Also why?

Reply:
Yes it should! I didn't include it because seqhash wasn't in Poly yet. Let me fix that.